django_docker_engine/docker_utils.py

```diff
@@ -140,6 +140,10 @@ def _kill_lru(self, need_to_free):
         lru_sorted = self._historian.sort_lru(container_ids)
         memory_freed = 0
         while memory_freed < need_to_free:
+            if len(lru_sorted) == 0:
+                logger.warn('No more containers to kill, but we still do not '
+                            'have the requested memory; Starting anyway!')
```

Comment:
What are the implications?

Reply:
If the limit-in-metadata that we check here closely reflects the actual system limit, then something else will quickly go wrong, but I don't have an easy reproducer for that: I can lower the allocation for Docker on Mac... but even at the minimum it's still a multiple of our most greedy tool

If the limit-in-metadata is artificially low, then it's really not a problem, though the next tool to start will evict this one.